### PR TITLE
ci: Remove CMake presets

### PR DESCRIPTION
**Pull request by nbolton**
Blocked by: #7740

Based on the discussion in: https://github.com/deskflow/deskflow/issues/7726#issuecomment-2419819619 and [Project Config](https://github.com/deskflow/deskflow/wiki/Project-Config)

### CODE REVIEW

File: res/dist/arch/PKGBUILD.in
Comment (nbolton) [inline res/dist/arch/PKGBUILD.in:1]: From: #7740

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -166,7 +166,11 @@ jobs:
       - name: Configure
         env:
           VCPKG_ROOT: ${{github.workspace}}/vcpkg
-        run: cmake -B build --preset=windows-release -DCMAKE_COMPILE_WARNING_AS_ERROR=ON
+        run: |
+          cmake -B build `
+            -G "Ninja" `
+            -DCMAKE_COMPILE_WARNING_AS_ERROR=ON `
+            -DCMAKE_TOOLCHAIN_FILE="$env:VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake"
```
Comment (nbolton) [inline .github/workflows/ci.yml:173]: Hmm, any ideas?
```
  C:\Program Files (x86)\WiX Toolset v3.14\bin\Light.exe -out D:\a\deskflow\deskflow\build\installer\bin\Release\Installer.msi -pdbout D:\a\deskflow\deskflow\build\installer\bin\Release\Installer.wixpdb -cultures:null -ext "C:\Program Files (x86)\WiX Toolset v3.14\bin\WixUtilExtension.dll" -ext "C:\Program Files (x86)\WiX Toolset v3.14\bin\WixUIExtension.dll" -ext "C:\Program Files (x86)\WiX Toolset v3.14\bin\WixFirewallExtension.dll" -contentsfile wix\obj\Release\Installer.wixproj.BindContentsFileListnull.txt -outputsfile wix\obj\Release\Installer.wixproj.BindOutputsFileListnull.txt -builtoutputsfile wix\obj\Release\Installer.wixproj.BindBuiltOutputsFileListnull.txt -wixprojectfile D:\a\deskflow\deskflow\build\installer\Installer.wixproj wix\obj\Release\pth266D54E62BB3A88CFDE192120250CA73\AppWelcome.wixobj wix\obj\Release\pth266D54E62BB3A88CFDE192120250CA73\AppInstallDirDlg.wixobj wix\obj\Release\pth266D54E62BB3A88CFDE192120250CA73\AppBrowseDlg.wixobj wix\obj\Release\pth266D54E62BB3A88CFDE192120250CA73\AppVe
D:\a\deskflow\deskflow\build\installer\Product.wxs(117): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\Qt6Core.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
D:\a\deskflow\deskflow\build\installer\Product.wxs(118): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\Qt6Gui.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
D:\a\deskflow\deskflow\build\installer\Product.wxs(119): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\Qt6Network.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
D:\a\deskflow\deskflow\build\installer\Product.wxs(120): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\Qt6Svg.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
D:\a\deskflow\deskflow\build\installer\Product.wxs(121): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\Qt6Widgets.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
D:\a\deskflow\deskflow\build\installer\Product.wxs(122): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\styles\qmodernwindowsstyle.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
D:\a\deskflow\deskflow\build\installer\Product.wxs(128): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\styles\qmodernwindowsstyle.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
D:\a\deskflow\deskflow\build\installer\Product.wxs(136): error LGHT0103: The system cannot find the file 'D:/a/deskflow/deskflow/build/bin\platforms\qwindows.dll'. [D:\a\deskflow\deskflow\build\installer\Installer.wixproj]
```

Here's the CMake preset that makes it work. I can't remember why this makes it work.
```
    {
      "name": "windows",
      "inherits": "base",
      "hidden": true,
      "generator": "Ninja",
      "cacheVariables": {
        "CMAKE_C_COMPILER": "cl.exe",
        "CMAKE_CXX_COMPILER": "cl.exe",
        "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
      },
      "architecture": {
        "value": "x64",
        "strategy": "external"
      },
      "toolset": {
        "value": "host=x64",
        "strategy": "external"
      }
    },
```

Comment (sithlord48) [reply .github/workflows/ci.yml:173]: Yeah bad hack in your packging  need the Qt Dir Set

File: BUILD.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ cmake -B build
 
 *Windows:*
 ```
-cmake -B build --preset=windows-release
```
Comment (sithlord48) [inline BUILD.md:33]: -G Ninja 

Windows and Linux users should use ninja in our  build examples

Comment (nbolton) [reply BUILD.md:33]: Isn't that what presets are for? :laughing: 

Whatever you think is best.

Comment (sithlord48) [reply BUILD.md:33]: sure but they may not want to use ninja ..

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -166,7 +166,11 @@ jobs:
           github-binarycache: true
 
       - name: Configure
-        run: cmake ${{ steps.vcpkg.outputs.vcpkg-cmake-config }} -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_COMPILE_WARNING_AS_ERROR=ON
+        run: |
+          cmake ${{ steps.vcpkg.outputs.vcpkg-cmake-config }} `
+            -B build -G Ninja `
+            -DCMAKE_BUILD_TYPE=Release `
+            -DCMAKE_COMPILE_WARNING_AS_ERROR=ON
```
Comment (nbolton) [inline .github/workflows/ci.yml:173]: I find it a bit easier to read when broken over multiple lines.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nbolton) [reply .github/workflows/ci.yml]: Yay, fixed.

**Review by sithlord48 — APPROVED** [review]

**Pull request merged**